plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Transform/CaseOfCase.hs

```diff
@@ -47,7 +72,18 @@ processTerm = \case
           mkIterApp
             ite
             [ cond
-            , (trueAnn, Delay trueAnn (Case ann true alts))
-            , (falseAnn, Delay falseAnn (Case ann false alts))
+            , (trueAnn, Delay trueAnn . CaseReduce.processTerm $ Case ann true alts)
+            , (falseAnn, Delay falseAnn . CaseReduce.processTerm $ Case ann false alts)
```

Comment:
TODO: document.

Reply:
I couldn't find test coverage for this logic. Did you also update the tests?